compiler-rt/lib/tysan/tysan_interceptors.cpp

```diff
@@ -99,6 +105,15 @@ INTERCEPTOR(char *, __strdup, const char *s) {
 }
 #endif // TYSAN_INTERCEPT___STRDUP
 
+#if TYSAN_INTERCEPT___STRNDUP
+INTERCEPTOR(char *, strndup, const char *s, SIZE_T slen) {
+  char *res = REAL(strndup)(s, slen);
+  if (res)
+    tysan_copy_types(res, const_cast<char *>(s), Min(internal_strlen(s), slen));
```

Comment:
Why do we need to cast away the `const`?

```
tysan_copy_types(const void *daddr, const void *saddr, uptr size) {
```

Reply:
indeed there is no need I did same as existing *strdup, maybe this call had a different signature before. Anyhow will update these later.